plugins/uefi-dbx/fu-dbxtool.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupd.h>
+#include <glib/gi18n.h>
+#include <glib-unix.h>
+#include <locale.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "fu-efivar.h"
+#include "fu-uefi-dbx-common.h"
+#include "fu-uefi-dbx-file.h"
+
+/* custom return code */
+#define EXIT_NOTHING_TO_DO		2
+
+static void
+fu_util_ignore_cb (const gchar *log_domain, GLogLevelFlags log_level,
+		   const gchar *message, gpointer user_data)
+{
+}
+
+int
+main (int argc, char *argv[])
+{
+	GPtrArray *checksums_system;
+	GPtrArray *checksums_update;
+	gboolean action_apply = FALSE;
+	gboolean action_list = FALSE;
+	gboolean force = FALSE;
+	gboolean verbose = FALSE;
+	gsize bufsz = 0;
+	g_autofree gchar *dbxfile = NULL;
+	g_autofree guint8 *buf_system = NULL;
+	g_autofree guint8 *buf_update = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_system = NULL;
+	g_autoptr(FuUefiDbxFile) dbx_update = NULL;
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GOptionContext) context = NULL;
+	const GOptionEntry options[] = {
+		{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+			/* TRANSLATORS: command line option */
+			_("Show extra debugging information"), NULL },
+		{ "list", 'l', 0, G_OPTION_ARG_NONE, &action_list,
+			/* TRANSLATORS: command line option */
+			_("List entries in dbx"), NULL },
```

Comment:
I think you should have two list commands - one for parsing the current dbx and one for parsing the new dbx

Reply:
Fixed, but keeping one command. If you specify `--dbx=` you list the contents of the file, otherwise you get the system values.